api/v1/helpers/verbs/utils.js

```diff
@@ -238,20 +240,41 @@ function handleAssociations(reqObj, inst, props, method) {
 } // handleAssociations
 
 /**
- * Generates sequelize options object with all the appropriate attributes
+ * Generates Sequelize options object with all the appropriate attributes
  * (fields) and includes, and taking virtual fields into account as well.
- * Always include the "id" field even if it was not explicitly requested.
+ * Include always:
+ *  The "id" field even if it was not explicitly requested.
+ *  All table foreign keys to avoid multiple association nested queries
+ *  failures.
  *
  * @param {Object} params - The request parameters
- * @returns {Object} - Sequelize options
+ * @param {Object} model - The helpers/nouns DB model
+ * @returns {Object} - option containing either attributes to be used
+ *  in Sequelize and foreign keys to be removed when send back to the client.
  */
-function buildFieldList(params) {
+function buildFieldList(params, model) {
   const opts = {};
   if (params.fields && params.fields.value) {
     opts.attributes = params.fields.value;
     if (!opts.attributes.includes('id')) {
       opts.attributes.push('id');
     }
+
+    /* Includes all model's FK for an eventual outer join. */
+    if (model) {
+      opts.fieldsToExclude = [];
+      const keys = Object.keys(model.attributes);
+      keys.forEach((key) => {
+        if (model.attributes && model.attributes[key].references) {
+          if (opts.attributes.indexOf(key) === NOT_FOUND) {
+            opts.attributes.push(key);
```

Comment:
what happened to only adding the key when necessary? (when multiple keys are references) I don't think we should be adding the fields to the db request for every model when it's only needed for Generators.

Reply:
How would you define "when necessary"? What happens if it is not a Generator and we got multiple keys? Should we check "if model.name === "generator""?

I am assuming this as a general workaround when we have sub-query using FK.